File: src/codes/Users_Manual/link/visualizer_calc.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Link;
~using AUTD3Sharp.Gain;
~using AUTD3Sharp.Modulation;
~using AUTD3Sharp.Utils;
var autd = Controller.Builder([new AUTD3(Vector3.Zero)]).Open(Visualizer.Builder());

var center = autd.Geometry.Center + new Vector3(0, 0, 150);
var g = new Focus(center);
autd.Send(g);

var points = new List<Vector3> { center };
var p = autd.Link.CalcField(points, autd.Geometry, Segment.S0, 0);
Console.WriteLine($"Acoustic pressure at ({center.X}, {center.Y}, {center.Z}) = ({p[0]})");

File: src/codes/Users_Manual/link/visualizer_calc.py
~import numpy as np
~from pyautd3 import AUTD3, Controller, Segment
~from pyautd3.gain import Focus
~from pyautd3.link.visualizer import Visualizer
autd: Controller[Visualizer] = (
    Controller[Visualizer].builder([AUTD3([0.0, 0.0, 0.0])]).open(Visualizer.builder())
)

center = autd.geometry.center + np.array([0, 0, 150])
g = Focus(center)
autd.send(g)

points = [center]
p = autd.link.calc_field(points, autd.geometry, Segment.S0, 0)
print(f"Acoustic pressure at ({center[0]}, {center[1]}, {center[2]}) = {p[0]}")